shub/deploy.py

```diff
@@ -189,8 +192,16 @@ def _get_pipfile_requirements():
         # Scrapy Cloud also doesn't support editable packages
         if 'editable' in v:
             del v['editable']
-    return open(convert_deps_to_pip(deps), 'rb')
-
+    return open(_add_sources(convert_deps_to_pip(deps), _sources=sources.encode()), 'rb')
+
+def _add_sources(_reqs_file, _sources):
+    tmp = tempfile.NamedTemporaryFile(delete=False, suffix="-requirements.txt")
```

Comment:
The temporary file should be deleted in the end of deploy step, I believe we shouldn't rely on an OS cleanup policy. As an option we could use a tracked named temporary file like it's used [here](https://github.com/pypa/pipenv/blob/dad1c2031e239df7f882c6ee38d5ce1fe2ca1a40/pipenv/utils.py#L1441)